packages/grid/backend/backend.dockerfile

```diff
@@ -1,94 +1,70 @@
 ARG PYTHON_VERSION="3.12"
-ARG TZ="Etc/UTC"
-
-# change to USER="syftuser", UID=1000 and HOME="/home/$USER" for rootless
-ARG USER="root"
-ARG UID=0
-ARG USER_GRP=$USER:$USER
-ARG HOME="/root"
-ARG APPDIR="$HOME/app"
+ARG UV_VERSION="0.1.41-r0"
+ARG TORCH_VERSION="2.3.0"
 
 # ==================== [BUILD STEP] Python Dev Base ==================== #
 
-FROM cgr.dev/chainguard/wolfi-base as python_dev
+FROM cgr.dev/chainguard/wolfi-base as syft_deps
 
 ARG PYTHON_VERSION
-ARG TZ
-ARG USER
-ARG UID
+ARG UV_VERSION
+ARG TORCH_VERSION
 
 # Setup Python DEV
-RUN --mount=type=cache,target=/var/cache/apk,sharing=locked \
-    apk update && \
-    apk upgrade && \
-    apk add build-base gcc tzdata python-$PYTHON_VERSION-dev-default py$PYTHON_VERSION-pip && \
-    ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
-# uncomment for creating rootless user
-# && adduser -D -u $UID $USER
-
-# ==================== [BUILD STEP] Install Syft Dependency ==================== #
-
-FROM python_dev as syft_deps
-
-ARG APPDIR
-ARG HOME
-ARG UID
-ARG USER
-ARG USER_GRP
-
-USER $USER
-WORKDIR $APPDIR
-ENV PATH=$PATH:$HOME/.local/bin
-
-# copy skeleton to do package install
-COPY --chown=$USER_GRP \
-    syft/setup.py \
-    syft/setup.cfg \
-    syft/pyproject.toml \
-    syft/MANIFEST.in \
-    syft/
-
-COPY --chown=$USER_GRP \
-    syft/src/syft/VERSION \
-    syft/src/syft/capnp \
-    syft/src/syft/
-
-# Install all dependencies together here to avoid any version conflicts across pkgs
-RUN --mount=type=cache,id=pip-$UID,target=$HOME/.cache/pip,uid=$UID,gid=$UID,sharing=locked \
-    pip install --user --default-timeout=300 torch==2.2.1 -f https://download.pytorch.org/whl/cpu/torch_stable.html && \
-    pip install --user pip-autoremove jupyterlab -e ./syft[data_science] && \
-    pip-autoremove ansible ansible-core -y
+RUN apk update && apk upgrade && \
+    apk add build-base gcc python-$PYTHON_VERSION-dev-default uv=$UV_VERSION
+
+WORKDIR /root/app
+
+ENV UV_HTTP_TIMEOUT=600
+
+# keep static deps separate to have each layer cached independently
+# if amd64 then we need to append +cpu to the torch version
+# uv issues: https://github.com/astral-sh/uv/issues/3437 & https://github.com/astral-sh/uv/issues/2541
+RUN --mount=type=cache,target=/root/.cache,sharing=locked \
+    uv venv && \
+    ARCH=$(arch | sed s/aarch64/arm64/ | sed s/x86_64/amd64/) && \
+    if [[ "$ARCH" = "amd64" ]]; then TORCH_VERSION="$TORCH_VERSION+cpu"; fi && \
+    uv pip install torch==$TORCH_VERSION --index-url https://download.pytorch.org/whl/cpu
+
+COPY syft/setup.py syft/setup.cfg syft/pyproject.toml ./syft/
+
+COPY syft/src/syft/VERSION ./syft/src/syft/
+
+RUN --mount=type=cache,target=/root/.cache,sharing=locked \
+    # remove torch because we already have the cpu version pre-installed
+    sed --in-place /torch==/d ./syft/setup.cfg && \
+    uv pip install -e ./syft[data_science,telemetry] && \
```

Comment:
I think we omit telemetry for the default docker build.

Currently, we mainly use telemetry for dev purposes (The telemetry in the codebase is also old, not sure if is functional)

maybe we could be keep it as part of start.sh, where we have a new env variable called TELEMETRY_ENABLED in backen-statefulset instead of having it in devmode.


```sh
if [[ ${TELEMETRY_ENABLED} == "True" ]];
then
    uv pip install ./syft[telemetry]
    
fi
```
Thoughts

Reply:
Let's avoid runtime pip installs because it creates race conditions that we miss out on - in this case a delayed DEV_MODE pip install resulted in seaweedfs race that got caught because of fast container startup time.

And yes, telemetry is old and we should definitely relook into it later. Happy to remove telemetry for now.